src/charm.py

```diff
@@ -30,6 +31,74 @@ def __init__(self, *args: typing.Any) -> None:
         super().__init__(*args)
         self.framework.observe(self.on.config_changed, self.reconciliation)
 
+    def _application_config(self) -> dict | None:
+        """Decode the value of the charm configuration application-config.
+
+        Returns:
+            The value of the charm configuration application-config.
+
+        Raises:
+            ReconciliationError: when application-config is invalid.
+        """
+        try:
+            config = self.model.config["application-config"]
+            if not config:
+                return None
+            application_config = json.loads(config)
+            if isinstance(application_config, dict):
+                return application_config
+            logger.error("Invalid application-config value: %s", repr(config))
+            raise ReconciliationError(
+                new_status=BlockedStatus(
+                    "Invalid application-config value, expecting an object in JSON"
+                )
+            )
+        except json.JSONDecodeError as exc:
+            raise ReconciliationError(
+                new_status=BlockedStatus("Invalid application-config value, expecting JSON")
+            ) from exc
+
+    def _spring_boot_port(self) -> int:
```

Comment:
Out of curiosity, why do we want the container port to be configurable? I don't see the advantage over using a default

Reply:
It's not that we want it to be configurable, Spring Boot allows it to be configurable. So if the user injects a configuration through "application-config" that changes the embedded webserver port we have 2 choices:
- Either ignore this parameter by removing the port configuration from the application-config
- Or follow the user configuration and adapt the charm according to it

I chose the later because changing what the user gave us as input could lead to confusion

Reply:
Makes sense. Thanks